dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.builder.BitstreamBuilder;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.builder.RequestItemBuilder;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MvcResult;
+
+/**
+ *
+ * @author Mark H. Wood <mwood@iupui.edu>
+ */
+public class RequestItemRepositoryIT
+        extends AbstractControllerIntegrationTest {
+    /** Where to find {@link RequestItem}s in the local URL namespace. */
+    public static final String URI_ROOT = REST_SERVER_URL
+            + RequestItemRest.CATEGORY + '/'
+            + RequestItemRest.NAME + 's';
+
+    @Autowired(required = true)
+    RequestItemConverter requestItemConverter;
+
+    @Autowired(required = true)
+    RequestItemService requestItemService;
+
+    @Before
+    public void init() {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context).withName(
+                "Parent Community").build();
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of findOne method, with an authenticated user.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testFindOneAuthenticated()
+            throws Exception {
+        System.out.println("findOne");
+
+        context.turnOffAuthorisationSystem();
+
+        // Create necessary supporting objects.
+        Collection collection = CollectionBuilder.createCollection(context, parentCommunity)
+                .build();
+        Item item = ItemBuilder.createItem(context, collection)
+                .build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .build();
+
+        // Create a request.
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        // Test:  can we find it?
+        String authToken = getAuthToken(admin.getEmail(), password);
+        final String uri = URI_ROOT + '/' + request.getToken();
+        getClient(authToken).perform(get(uri))
+                   .andExpect(status().isOk()) // Can we find it?
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$", Matchers.is(
+                       RequestCopyMatcher.matchRequestCopy(request))));
+
+        // Clean up.
+        bitstream.setDeleted(true);
+        context.restoreAuthSystemState();
```

Comment:
I don't believe the cleanup is needed...Builders should cleanup after themselves.

Also, please move the `context.restoreAuthSystemState()` up to *before* the `getClient()` call.  This just helps us ensure that we are always turning auth back on BEFORE tests run.   In general, turn the auth back on as soon as you can, which usually is right after you create any test content you need.

Reply:
Right.  Done.